aeron/publication.go

```diff
@@ -162,182 +157,175 @@ func (pub *Publication) Position() int64 {
 	termOffset := rawTail & 0xFFFFFFFF
 	termId := logbuffer.TermID(rawTail)
 
-	return computeTermBeginPosition(termId, pub.positionBitsToShift, pub.initialTermID) + termOffset
+	return computeTermBeginPosition(termId, pub.positionBitsToShift, pub.initialTermID) + termOffset, nil
 }
 
 // Offer is the primary send mechanism on Publication
-func (pub *Publication) Offer(buffer *atomic.Buffer, offset int32, length int32, reservedValueSupplier term.ReservedValueSupplier) int64 {
-
-	newPosition := PublicationClosed
+func (pub *Publication) Offer(buffer *atomic.Buffer, offset int32, length int32, reservedValueSupplier term.ReservedValueSupplier) (int64, error) {
```

Comment:
I think the use of position codes as offer responses was consistent with Java/C++. One big reason is that BackPressured and many other return codes aren't errors, they're expected. I think if you checked Java, Offer() is non-blocking and will never throw.

Reply:
Yes, I saw that.  I was thinking the idiomatic Go version of return codes was public named errors and use of errors.Is(err, BackPressuredErr).  However, even within the Go community, this is not universal.  I'll change it back for consistency/backwards compatibility.

We did have a few panics due to bad input (length negative or above the max).  They correspond with Java throws.  I'm sure those are rarely thrown in practice, but we should convert those to error returns here, right?

Reply:
Hm, I'm trying to figure out if panic'ing in those scenarios is OK. We don't ever expect them to happen. I would tend towards continuing to panic on these and leave the interface clean.

Reply:
Ok, will do